### ISSUE TITLE
pkrepo.manage does not detect keyid options changes

### ISSUE THREAD

**Issue by skateinmars** [saltstack/salt#9046]
Steps to reproduce:
- Write a state without GPG keys informations

```
nginx_repo:
  pkgrepo.managed:
    - humanname: Nginx repository
    - name: deb http://nginx.org/packages/debian/ wheezy nginx
    - dist: wheezy
    - file: /etc/apt/sources.list.d/nginx.list
    - require_in:
      - pkg: nginx
```

This state will succeed. Adding a nginx pkg.installed state will correctly fail as no key is present.
- Change your state to add GPG key infos:

```
nginx_repo:
  pkgrepo.managed:
    - humanname: Nginx repository
    - name: deb http://nginx.org/packages/debian/ wheezy nginx
    - dist: wheezy
    - file: /etc/apt/sources.list.d/nginx.list
    - keyid: 7BD9BF62
    - keyserver: keyserver.ubuntu.com
    - require_in:
      - pkg: nginx
```

salt-call state.highstate will report:

```
[INFO    ] Executing state pkgrepo.managed for deb http://nginx.org/packages/debian/ wheezy nginx
[INFO    ] Package repo 'deb http://nginx.org/packages/debian/ wheezy nginx' already configured
```

And no key will be added. I'm not sure this should be considered a bug though?

This was tested using salt version 0.17.2.

**Comment by notpeter** [comment]
This is still unfixed 18 months later.  Pkgrepo.managed does not verify keyserver/keyid if the sources.list.d file is correct.

I ran into this because of a transient problem with a keys.gnupg.net mirror. Once the sources.list entry was correct, it never again tried to retrieve the keys.

**Comment by jplatte** [comment]
Is it a big problem to check whether the GPG key is imported independently of the state of the repo file? I would really like to see this fixed.

**Comment by jplatte** [comment]
Maybe that bot should be taught to check for labels like 'High Severity'...

**Comment by skateinmars** [comment]
AFAIK this is still relevant?

**Comment by jplatte** [comment]
This is a duplicate of #4438 and thus should have been closed with #53537 AFAICT.

**Comment by sagetherage** [comment]
duplicate and fixed with #53537 closing
